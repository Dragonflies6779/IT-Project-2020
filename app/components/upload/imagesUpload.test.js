var React = require('react');
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import { screen ,getByLabelText,
  getByText,
  getByTestId,
  queryByTestId,
  toBeDisabled} from '@testing-library/dom'
var Upload = require('./imagesUpload.js');

it('renders correctly', () => {
  const tree = renderer.create(<Upload/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('The button is not disabled and works', () => {
  const utils = render(<Upload />)
  const input = utils.getAllByRole('button')
  expect(input[0].hasAttribute('disabled')).toBe(false)
})

it('Check if there a file upload input form', () => {
  const utils = render(<Upload />)
  const input = utils.getByRole('progressbar')
  expect(input.value).toBe(0)
})
