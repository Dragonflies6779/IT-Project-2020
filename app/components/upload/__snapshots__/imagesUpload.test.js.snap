// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <progress
    max="100"
  />
  <br />
  <input
    onChange={[Function]}
    type="file"
  />
  <button
    classname="imagesUploadButton"
    onClick={[Function]}
  >
    Upload
  </button>
</div>
`;
